lisa/tools/virtualclient.py

```diff
@@ -0,0 +1,275 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import csv
+import json
+from datetime import datetime
+from decimal import Decimal
+from pathlib import Path
+from typing import Any, List, Tuple
+
+from lisa import notifier
+from lisa.base_tools import Uname, Wget
+from lisa.executable import Tool
+from lisa.messages import VCMetrics
+from lisa.operating_system import Debian
+from lisa.util import LisaException
+from lisa.util.process import Process
+
+from .chmod import Chmod
+from .chown import Chown
+from .git import Git
+from .ln import Ln
+from .parted import Parted
+from .unzip import Unzip
+from .whoami import Whoami
+
+
+class VirtualClientTool(Tool):
```

Comment:
The class should be split into two parts, one is to operate vc on the single node, called like `VirtualClient`. Another one is to run workload on one or more nodes. It passes in multiple `VirtualClient` tools.

Reply:
I don't see other tools design like this, could you please tell me the reason, thanks.

Reply:
Other tools operate on one node. But the VirtualClient needs to operate on multiple nodes together. For NTTTCP, the operations of multiple nodes in test case level. But we want to abstract vc tools for general use, so it needs to be supported at the tool level.